nucliadb/src/nucliadb/search/search/chat/prompt.py

```diff
@@ -695,20 +696,31 @@ async def conversation_prompt_context(
 
                 attachments: List[resources_pb2.FieldRef] = []
                 if conversational_strategy.full:
+                    extracted_text = await field_obj.get_extracted_text()
                     for current_page in range(1, cmetadata.pages + 1):
                         conv = await field_obj.db_get_value(current_page)
+
                         for message in conv.messages:
-                            text = message.content.text.strip()
-                            pid = f"{rid}/{field_type}/{field_id}/{message.ident}/0-{len(message.content.text) + 1}"
+                            ident = message.ident
+                            if extracted_text is not None:
+                                text = extracted_text.split_text.get(ident, message.content.text.strip())
+                            else:
+                                text = message.content.text.strip()
+                            pid = f"{rid}/{field_type}/{field_id}/{ident}/0-{len(message.content.text) + 1}"
```

Comment:
Shoudn't this be:
```suggestion
                            pid = f"{rid}/{field_type}/{field_id}/{ident}/0-{len(text) - 1}"
```
?